### PR TITLE
Use ace editor for admin restore page

### PR DESCRIPTION
**Pull request by proteusvacuum**
Uses the ace editor on the admin restore page so that it doesn't try and render all the XML at once. This rendered a 20MB restore file for me immediately, and comes with its own search box.

Previous attempts at making this page usable are: `hide_xml`, and `download=true`. This is a combination of all of them!

### CODE REVIEW

File: corehq/apps/hqadmin/templates/hqadmin/admin_restore.html [hunk 1]
```diff
@@ -83,7 +86,7 @@ <h3>Response Status Code: {{ status_code }}</h3>
                         <div class="tab-content">
                             <div role="tabpanel" class="tab-pane active" id="payload-tab">
                                 {% if not hide_xml %}
-                                    <pre id="payload">{{ payload }}</pre>
+                                    <pre id="payload" data-payload="{{ payload }}"></pre>
```
Comment (orangejenny) [inline corehq/apps/hqadmin/templates/hqadmin/admin_restore.html]: Should this be `{% html_attr payload %}` to take care of potential special characters?

File: corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
-/* globals hqDefine */
+/* globals hqDefine ace */
 hqDefine('hqadmin/js/admin_restore', function () {
     $(function() {
         $("#timingTable").treetable();
+        var element = document.getElementById("payload");
+        var editor = ace.edit(
```
Comment (orangejenny) [inline corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js]: How much are these options going to vary on different pages? Is it too soon to make a standard set of options?

**Comment by orangejenny** [comment]
this is marvelous

File: corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js (hunk 2 above)
Comment (proteusvacuum) [reply corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js]: these are totally different to the other use, so "a lot".

File: corehq/apps/hqadmin/templates/hqadmin/admin_restore.html (hunk 1 above)
Comment (proteusvacuum) [reply corehq/apps/hqadmin/templates/hqadmin/admin_restore.html]: probably. updating.

**Review by orangejenny — APPROVED** [review]

**Review by pr33thi — APPROVED** [review]

**Pull request merged**